routing/graph.go

```diff
@@ -41,30 +41,29 @@ type CachedGraph struct {
 var _ routingGraph = (*CachedGraph)(nil)
 
 // NewCachedGraph instantiates a new db-connected routing graph. It implicitly
-// instantiates a new read transaction.
+// instantiates a new read transaction and returns a cleanup function.
 func NewCachedGraph(sourceNode *channeldb.LightningNode,
-	graph *channeldb.ChannelGraph) (*CachedGraph, error) {
+	graph *channeldb.ChannelGraph) (*CachedGraph, func(), error) {
 
 	tx, err := graph.NewPathFindTx()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	cleanUp := func() {}
```

Comment:
Added the comment to the commit message. It's for being able to use the `routingGraph` externally and being able to clean up there. Otherwise this lead to blocking due to the opened db transaction.

Reply:
Wasn't that cleanup possible previously by just calling `Close` on the returned `*CachedGraph` value?